### PR TITLE
Fix NullReferenceException in ChangeDeclarations

### LINKED ISSUE #159: NullReferenceException when computing style
### Prerequisites

- [X] Can you reproduce the problem in a [MWE](https://en.wikipedia.org/wiki/Minimal_working_example)?
- [X] Are you running the latest version of AngleSharp.Css?
- [X] Did you check the FAQs to see if that helps you?
- [X] Are you reporting to the correct repository? (there are multiple AngleSharp libraries, e.g., `AngleSharp.Xml` for Xml support)
- [X] Did you perform a search in the issues?

### Description

AngleSharp fails to compute style for `<div style=""></div>` with the latest preview `1.0.0-beta.122`

````
System.NullReferenceException: Object reference not set to an instance of an object.
   at AngleSharp.Css.Dom.CssStyleDeclaration.ChangeDeclarations(IEnumerable`1 decls, Predicate`1 defaultSkip, Func`3 removeExisting)
   at AngleSharp.Css.Dom.CssStyleDeclaration.SetDeclarations(IEnumerable`1 decls)
   at AngleSharp.Css.StyleCollectionExtensions.ComputeCascadedStyle(IStyleCollection styles, IElement element, ICssStyleDeclaration parent)
   at AngleSharp.Css.StyleCollectionExtensions.ComputeDeclarations(IStyleCollection styles, IElement element, String pseudoSelector)
   at AngleSharp.Dom.WindowExtensions.GetComputedStyle(IWindow window, IElement element, String pseudo)
   at Program.<Main>$(String[] args) in D:\source\ConsoleApp4\ConsoleApp4\Program.cs:line 16
   at Program.<Main>(String[] args)
````

### Steps to Reproduce

````xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net8.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="AngleSharp" Version="1.1.1-beta.386" />
    <PackageReference Include="AngleSharp.Css" Version="1.0.0-beta.122" />
  </ItemGroup>

</Project>
````

````c#
using AngleSharp.Css;
using AngleSharp;
using AngleSharp.Io;
using AngleSharp.Dom;

var configuration = new AngleSharp.Configuration()
            .WithDefaultLoader()
            .WithRenderDevice(new DefaultRenderDevice() { ViewPortHeight = 100, ViewPortWidth = 100 });

var context = BrowsingContext.New(configuration);
using var response = VirtualResponse.Create(request => request.Content("""
    <div style="transform-origin: bottom"></div>
    """));
var document = await context.OpenAsync(response, CancellationToken.None);
var element = document.QuerySelector("div");
var style = document.DefaultView.GetComputedStyle(element); // throw NRE
````

### Expected Behavior

No exception

### Actual Behavior

It throws a NullReferenceException

### Possible Solution / Known Workarounds

_No response_

### PR DESCRIPTION
**Pull request by meziantou**
# Types of Changes

## Prerequisites

Please make sure you can check the following two boxes:

- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project

## Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [x] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
- [ ] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

## Description

Fix #159

### CODE REVIEW

File: src/AngleSharp.Css/Dom/Internal/CssStyleDeclaration.cs [hunk 1]
```diff
@@ -377,6 +377,9 @@ private void RemovePropertyByName(String propertyName)
 
         private void ChangeDeclarations(IEnumerable<ICssProperty> decls, Predicate<ICssProperty> defaultSkip, Func<ICssProperty, ICssProperty, Boolean> removeExisting)
         {
+            if (decls == null)
```
Comment (FlorianRappl) [inline src/AngleSharp.Css/Dom/Internal/CssStyleDeclaration.cs]: Looks good to me - can we have this `decls is null` and with a block statement? Thanks!

```
if (decls is null)
{
    return;
}
```

File: src/AngleSharp.Css/Dom/Internal/CssStyleDeclaration.cs [hunk 2]
```diff
@@ -377,6 +377,9 @@ private void RemovePropertyByName(String propertyName)
...
+                return;
```
Comment (meziantou) [inline src/AngleSharp.Css/Dom/Internal/CssStyleDeclaration.cs]:
```suggestion
            if (decls is null)
            {
                return;
            }
```

File: src/AngleSharp.Css/Dom/Internal/CssStyleDeclaration.cs (hunk 1 above)
Comment (meziantou) [reply src/AngleSharp.Css/Dom/Internal/CssStyleDeclaration.cs]: done!

**Review by FlorianRappl — APPROVED** [review]
LGTM - thanks!

**Pull request merged**